content/solids/functions.ts

```diff
@@ -10,7 +10,141 @@ import {VoxelPainter} from './components/voxel-painter';
 import './components/voxel-painter';
 import '../shared/components/solid';
 import '../polyhedra/components/polyhedron';
+import {BLUE, GREEN, GREY, LIME, RED, YELLOW} from '../shared/constants';
 
+export async function templeFilling($step: Step) {
+  const $voxelPainter = $step.$('x-voxel-painter') as VoxelPainter;
+  // await $voxelPainter.ready();
+
+  handleLayers($voxelPainter, 0);
+}
+
+type VoxelData = {
+  locs: Array<[number, number, number]>,
+  color: string
+};
+
+const layers: VoxelData[] = [
+  {locs: [], color: ''},
+  {
+    locs: [
+      [-1.0, -2.0, -2.0], [0.0, -2.0, -2.0], [1.0, -2.0, -2.0], [2.0, -2.0, -2.0],
```

Comment:
Maybe run a replace `.0` => ` ` for the entire file!
(JavaScript has no integer vs float types, so it just increases the file size)

Reply:
Done